src/hooks/useToggleNode.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+type onClickProp = (e: React.KeyboardEvent | React.MouseEvent) => void;
+type onToggleProp = (e: React.KeyboardEvent | React.MouseEvent) => void;
+
+type onToggleNodeProps = {
+    onClick?: onClickProp;
+    onKeyUp?: React.KeyboardEventHandler;
+    disabled?: boolean;
+    ariaSelected?: boolean;
+    tabIndex?: number;
+};
+
+export const onToggleNode = (
+    onToggle: onToggleProp,
+    onClick?: onClickProp,
+    isDisabled = false,
+    tabIndex = 0
+): onToggleNodeProps => {
+    const handleKeyUp = (e: React.KeyboardEvent) => {
+        if (e.code === 'Space') {
+            onToggle(e);
+        }
+
+        if (e.code === 'Enter') {
+            e.preventDefault();
+            if (onClick && typeof onClick === 'function' && !isDisabled) {
```

Comment:
Maybe you can remove `typeof onClick === 'function'` or we should add it when we call `onToggle`